packages/bsp/common/etc/update-motd.d/10-armbian-header

```diff
@@ -14,8 +14,8 @@ MOTD_DISABLE=""
 
 [[ -f /etc/armbian-release ]] && . /etc/armbian-release
 [[ -f /etc/armbian-distribution-status ]] && . /etc/armbian-distribution-status
-[[ -f /etc/lsb-release && -f /etc/armbian-distribution-status ]] && DISTRIBUTION_CODENAME=$(cat /etc/lsb-release | grep CODENAME | cut -d"=" -f2) && DISTRIBUTION_STATUS=$(cat /etc/armbian-distribution-status | grep $DISTRIBUTION_CODENAME | cut -d"=" -f2)
-[[ -f /etc/os-release && -f /etc/armbian-distribution-status ]] && DISTRIBUTION_CODENAME=$(cat /etc/os-release | grep VERSION_CODENAME | cut -d"=" -f2) && DISTRIBUTION_STATUS=$(cat /etc/armbian-distribution-status | grep $DISTRIBUTION_CODENAME | cut -d"=" -f2)
+[[ -f /etc/lsb-release && -f /etc/armbian-distribution-status ]] && DISTRIBUTION_CODENAME=$(grep CODENAME /etc/lsb-release | cut -d"=" -f2) && DISTRIBUTION_STATUS=$(grep $DISTRIBUTION_CODENAME /etc/armbian-distribution-status | cut -d"=" -f2)
+[[ -f /etc/os-release && -f /etc/armbian-distribution-status ]] && DISTRIBUTION_CODENAME=$(grep VERSION_CODENAME /etc/os-release | cut -d"=" -f2) && DISTRIBUTION_STATUS=$(grep $DISTRIBUTION_CODENAME /etc/armbian-distribution-status | cut -d"=" -f2)
```

Comment:
This is not quite the same, and this script now hangs during login for me.

Reply:
The problem is that "VERSION_CODENAME" isn't found in `/etc/os-release`, so it ends up doing `grep /etc/armbian-distribution-status` without any pattern, meaning it tries to read from stdin instead...

Reply:
This issue triggers for me on the Debian SID image for la frite, BTW.

A quick-fix is to add quotes around $DISTRIBUTION_CODENAME... It prevents the hanging, but the result isn't perfect; we now end up reporting an unsupported userspace. I'm not sure if that was always the case, but it seems wrong to me...

@neheb: ^

Reply:
This seems to fix the problem properly, reporting the Sid codename:
```patch
 [[ -f /etc/armbian-release ]] && . /etc/armbian-release
 [[ -f /etc/armbian-distribution-status ]] && . /etc/armbian-distribution-statu
-[[ -f /etc/os-release && -f /etc/armbian-distribution-status ]] && DISTRIBUTION_CODENAME=$(grep VERSION_CODENAME /etc/os-release | cut -d"=" -f2) && DISTRIBUTION_STATUS=$(grep $DISTRIBUTION_CODENAME /etc/armbian-distribution-status | cut -d"=" -f2)
+[[ -f /etc/armbian-distribution-status ]] && DISTRIBUTION_CODENAME=$(lsb_release -c | cut -d":" -f2 | tr -d "\t") && DISTRIBUTION_STATUS=$(grep "$DISTRIBUTION_CODENAME" /etc/armbian-distribution-status | cut -d"=" -f2)
 [[ -f /etc/default/armbian-motd ]] && . /etc/default/armbian-motd
```

Reply:
Sounds good.

Reply:
Reading the commit history here, it seems reading os-release was intentional, in 0a2fdc4f0b9f693ca0f7947d2e2fa539a9c73d6a by @bl4cki. However, it doesn't seem to work on Sid (or Bookwork), because they use the 12.2 release of the base-files package, which no longer contains the `DISTRIBUTION_CODENAME`-entry that was there in the 11.1 release.

Reply:
Fixed in !4144